common/src/chain/transaction/printout.rs

```diff
@@ -0,0 +1,382 @@
+            fmt_dest(p.staker()),
+            fmt_vrf(p.vrf_public_key()),
+            fmt_dest(p.decommission_key()),
+            p.margin_ratio_per_thousand().into_percentage_str(),
+            fmt_ml(&p.cost_per_block())
+        )
+    };
+    let fmt_tkn_supply = |s: &TokenTotalSupply, d: u8| match s {
+        TokenTotalSupply::Fixed(v) => format!("Fixed({})", v.into_fixedpoint_str(d)),
+        TokenTotalSupply::Lockable => "Lockable".to_string(),
+        TokenTotalSupply::Unlimited => "Unlimited".to_string(),
+    };
+    let fmt_tkn_frzble = |f: &IsTokenFreezable| match f {
+        IsTokenFreezable::No => "Yes".to_string(),
+        IsTokenFreezable::Yes => "No".to_string(),
+    };
+    let fmt_tkn_iss = |iss: &TokenIssuance| {
+        match iss {
+        TokenIssuance::V1(iss1) => format!(
+            "TokenIssuance(Ticker({}), Decimals({}), MetadataUri({}), TotalSupply({}), Authority({}), IsFreezable({}))",
+            String::from_utf8_lossy(&iss1.token_ticker),
+            iss1.number_of_decimals,
+            String::from_utf8_lossy(&iss1.metadata_uri),
+            fmt_tkn_supply(&iss1.total_supply, iss1.number_of_decimals),
+            fmt_dest(&iss1.authority),
+            fmt_tkn_frzble(&iss1.is_freezable)
+        ),
+    }
+    };
+    let fmt_nft_iss = |iss: &NftIssuance| match iss {
+        NftIssuance::V0(iss1) => {
+            let md = &iss1.metadata;
+            let creator = match &md.creator {
+                Some(c) => hex::encode(c.public_key.encode()).to_string(),
+                None => "Unspecified".to_string(),
+            };
+            format!(
+                "Create({}), Name({}), Description({}), Ticker({}), IconUri({}), AdditionalMetaData({}), MediaUri({}), MediaHash(0x{})",
+                creator,
+                String::from_utf8_lossy(&md.name),
+                String::from_utf8_lossy(&md.description),
+                String::from_utf8_lossy(&md.ticker),
+                String::from_utf8_lossy(md.icon_uri.as_ref().as_ref().unwrap_or(&vec![])),
+                String::from_utf8_lossy(
+                    md.additional_metadata_uri.as_ref().as_ref().unwrap_or(&vec![])
+                ),
+                String::from_utf8_lossy(
+                    md.media_uri.as_ref().as_ref().unwrap_or(&vec![])
+                ),
+                hex::encode(&md.media_hash),
+
+            )
+        }
+    };
+
+    for input in tx.inputs() {
+        writeln!(&mut result, "- {input:?}").expect("Writing to a memory buffer should not fail");
+    }
+
+    writeln!(
+        &mut result,
+        "=== END OF INPUTS ===\n=== BEGIN OF OUTPUTS ==="
+    )
+    .expect("Writing to a memory buffer should not fail");
+
+    for output in tx.outputs() {
+        let s = match output {
+            TxOutput::Transfer(val, dest) => {
+                let val_str = fmt_val(val);
+                format!("Transfer({}, {val_str})", fmt_dest(dest))
+            }
```

Comment:
The output printout could be taken out to a separate function in case someone wants to print outputs separately. Similarly with inputs.